internal/transport/noop.go

```diff
@@ -0,0 +1,33 @@
+package transport
+
+import "crypto/tls"
+
+// Noop is a Transporter that does nothing. This Transport is used to enable a
+// "local only" dispatch mode.
+type Noop struct{}
+
+func NewNoopTransport() (*Noop, error) {
```

Comment:
I would add here at least one info/warning log message saying that we are using "noop" transport "protocol" and nothing will be sent or received.